shared/shared.go

```diff
@@ -171,3 +173,55 @@ func DecodeSliceOfByteSliceWithLimit(d *scale.Decoder, outerLimit, innerLimit ui
 
 	return result, total, nil
 }
+
+// FindSubmitPowNonce finds the nonce that solves the PoW challenge.
+func FindSubmitPowNonce(
+	ctx context.Context,
+	powChallenge, poetChallenge, nodeID []byte,
+	difficulty uint,
+) (uint64, error) {
+	var hash []byte
+	for nonce := uint64(0); ; nonce++ {
+		select {
+		case <-ctx.Done():
+			return 0, ctx.Err()
+		default:
+		}
+
+		hash := CalcSubmitPowHash(powChallenge, poetChallenge, nodeID, hash, nonce)
+		if CheckLeadingZeroBits(hash, difficulty) {
+			return nonce, nil
+		}
+	}
+}
+
+// CalcSubmitPowHash calculates the hash for the Submit PoW.
+// The hash is ripemd256(sha256(powChallenge || poetChallenge || nodeID || nonce).
```

Comment:
> ripemd256(sha256(powChallenge || poetChallenge || nodeID || nonce)

it doesn't seem accurate, is it outdated? it looks like sha256 should be removed (also in other places)

Reply:
thx